### ISSUE TITLE
atomic::compare_exchange_strong()

### ISSUE THREAD

**Issue by mrzv** [illuhad/hipSYCL#174]
I noticed that `atomic::compare_exchange_strong()` is marked as unimplemented. Doesn't the following implementation work, or am I missing something?
```
template<class T, cl::sycl::access::address_space addressSpace>
HIPSYCL_KERNEL_TARGET
bool
cl::sycl::atomic<T,addressSpace>::
compare_exchange_strong(T &expected, T desired,
                        memory_order successMemoryOrder,
                        memory_order failMemoryOrder) volatile
{
#ifdef __HIPSYCL_DEVICE_CALLABLE__
    T old = expected;
    expected = atomicCAS(_ptr, expected, desired);
    return old == expected;
#else
    return detail::invalid_host_call_dummy_return<T>();
#endif
}   
```

**Comment by illuhad** [comment]
I think this is correct. I'm not sure if there is maybe another subtle reason why I left this unimplemented (e.g. small differences in the required memory model), but as a start this looks fine.

**Comment by Alkeryn** [comment]
I'm really waiting on that one as it is kind of the only way to add atomic operation for unsuported stuff.

**Comment by illuhad** [comment]
@Alkeryn please feel free to submit a PR ;)

**Comment by illuhad** [comment]
Fixed by PR #209
